packages/api/src/user.js

```diff
@@ -569,6 +569,43 @@ export async function userPinsGet (request, env) {
   }, { headers })
 }
 
+/**
+ *  List a user's pins regardless of the token used.
+ *  As we don't want to scope the Pinning Service API to users
+ *  we need a new endpoint as an umbrella.
+ *
+ * @param {import('./user').AuthenticatedRequest} request
+ * @param {import('./env').Env} env
+ * @param {import('./index').Ctx} ctx
+ * @return {Promise<JSONResponse>}
+ */
+export async function userPinDelete (request, env, ctx) {
+  // @ts-ignore
```

Comment:
Good catch!

The issue here is that `AuthenticatedRequest` extends `Request` while it should really extend `import('itty-router').Request` that does define the `param` property.

The challenge here is that some of the property definitions do not quite match with `Request`.
If you look [here](https://github.com/kwhitley/itty-router/blob/v2.x/src/itty-router.d.ts#L19), most of the properties are optional, which I don't think it is necessarely accurate. And updating that type would cause lots of type errors across the whole package.
From a quick look at the code in `itty-router` I think a better type would be extending `Request` (Fetch Api one) and adding `params` and `query`.

That said, I wonder if it's better to tackle this in its own ticket to gauge what the best approach here, and given it's a common pattern to use `request.parameters` (with `@ts-ignore`) across the package to leave it as in here?
Actually I'll update the code to check if `params` exists given I think it can be udefined.


--- 
Possible fixes:
  1. updating the type to extend from `ittyrouter.Request`  **or** 
  2. create our own `Request` one with the additional properties and have `Authenticated Request` to extend from it) .
  3 something else

Reply:
Cool, yep that’s a great shout to address the wider issue in a separate ticket.
And also a great shout to check if Paramus exist since we should consider it optional if not typed correctly 👍